test/jasmine/tests/download_test.js

```diff
@@ -59,8 +70,55 @@ describe('Plotly.downloadImage', function() {
     it('should create link, remove link, accept options', function(done) {
         downloadTest(gd, 'svg', done);
     }, LONG_TIMEOUT_INTERVAL);
-});
 
+    it('should produce the right SVG output in IE', function(done) {
+        // mock up IE behavior
+        Lib.isIE = function() { return true; };
+        slzProto.serializeToString = function() {
+            return serializeToString.apply(this, arguments)
+                .replace(/(\(#)([^")]*)(\))/gi, '(\"#$2\")');
+        };
+        var savedBlob;
+        navigator.msSaveBlob = function(blob) { savedBlob = blob; };
+
+        var expectedStart = '<svg class=\'main-svg\' xmlns=\'http://www.w3.org/2000/svg\' xmlns:xlink=\'http://www.w3.org/1999/xlink\'';
+        var plotClip = /clip-path='url\("#clip[0-9a-f]{6}xyplot"\)/;
+        var legendClip = /clip-path=\'url\("#legend[0-9a-f]{6}"\)/;
+
+        Plotly.plot(gd, textchartMock.data, textchartMock.layout)
+        .then(function(gd) {
+            savedBlob = undefined;
+            return Plotly.downloadImage(gd, {
+                format: 'svg',
+                height: 300,
+                width: 300,
+                filename: 'plotly_download'
+            });
+        })
+        .then(function() {
+            expect(savedBlob).toBeDefined();
+            if(savedBlob === undefined) return;
```

Comment:
Perhaps

```js
if(saveBlob === undefined) {
  fail('undefined save blob');
}
```

would be more readable than an `expect` + early `return` sequence.